### PR TITLE
Modernize a bit

### PR DESCRIPTION
**Pull request by Goooler**
## Description

Addressed from #2644.

## Type of change

- [x] General change (non-breaking change that fixes typos/grammar, or adds additional content that isn't a bug or a feature)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -79,10 +77,10 @@ allprojects {
         }
     }
 
-    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
-        kotlinOptions {
-            freeCompilerArgs += "-opt-in=kotlin.RequiresOptIn"
```
Comment (Goooler) [inline build.gradle:84]: https://kotlinlang.org/docs/whatsnew17.html#stable-opt-in-requirements

File: build.gradle [hunk 2]
```diff
@@ -79,10 +77,10 @@ allprojects {
...
-        }
+    tasks.matching {
+        it.name.contains("transformClassesWithHiddenApiRefine")
+    }.configureEach {
+        notCompatibleWithConfigurationCache("https://github.com/RikkaApps/HiddenApiRefinePlugin/issues/9")
```
Comment (Goooler) [inline build.gradle:83]: https://github.com/gradle/gradle/issues/19765#issuecomment-1026728095

**Comment by Goooler** [comment]
```
Verification succesful
/usr/bin/cp build/outputs/apk/lawnWithQuickstep/debug/Lawnchair 12.1.0 Dev (#1002).apk build/outputs/apk/lawnWithQuickstep/debug/Lawnchair 12.1.0 Dev (#1002)-aligned.apk
/usr/local/lib/android/sdk/build-tools/29.0.3/apksigner sign --ks build/outputs/apk/lawnWithQuickstep/debug/signingKey.jks --ks-key-alias  --ks-pass pass: --out build/outputs/apk/lawnWithQuickstep/debug/Lawnchair 12.1.0 Dev (#1002)-signed.apk build/outputs/apk/lawnWithQuickstep/debug/Lawnchair 12.1.0 Dev (#1002)-aligned.apk
Failed to load signer "signer #1"
java.io.IOException: Tag number over 30 is not supported
	at java.base/sun.security.util.DerValue.<init>(DerValue.java:442)
	at java.base/sun.security.util.DerValue.<init>(DerValue.java:487)
	at java.base/sun.security.pkcs12.PKCS12KeyStore.engineLoad(PKCS12KeyStore.java:2012)
	at java.base/sun.security.util.KeyStoreDelegator.engineLoad(KeyStoreDelegator.java:221)
	at java.base/java.security.KeyStore.load(KeyStore.java:1473)
	at com.android.apksigner.SignerParams.loadKeyStoreFromFile(SignerParams.java:353)
	at com.android.apksigner.SignerParams.loadPrivateKeyAndCertsFromKeyStore(SignerParams.java:239)
	at com.android.apksigner.SignerParams.loadPrivateKeyAndCerts(SignerParams.java:181)
	at com.android.apksigner.ApkSignerTool.sign(ApkSignerTool.java:277)
	at com.android.apksigner.ApkSignerTool.main(ApkSignerTool.java:83)
Error: The process '/usr/local/lib/android/sdk/build-tools/29.0.3/apksigner' failed with exit code 2
```

**Comment by Goooler** [comment]
Seems `r0adkll/sign-android-release@v1` haven't supported SDK 30 and above, see https://github.com/r0adkll/sign-android-release/issues/68.

**Comment by msfjarvis** [comment]
@Goooler can you rebase this and look into the failing CI? Looks like the signing workaround didn't fix anything.

**Comment by Goooler** [comment]
> can you rebase this and look into the failing CI? Looks like the signing workaround didn't fix anything.

Maybe we need to write sign info into property file, like this https://github.com/zhaobozhen/LibChecker/blob/22e2be4f5d42d9475a014f52879e55e434880cfa/.github/workflows/android.yml#L64-L72

**Comment by msfjarvis** [comment]
> > can you rebase this and look into the failing CI? Looks like the signing workaround didn't fix anything.
> 
> Maybe we need to write sign info into property file, like this [zhaobozhen/LibChecker@`22e2be4`/.github/workflows/android.yml#L64-L72](https://github.com/zhaobozhen/LibChecker/blob/22e2be4f5d42d9475a014f52879e55e434880cfa/.github/workflows/android.yml#L64-L72)

I'd be fine with that 👍🏼

File: .github/workflows/build_debug_apk.yml [hunk 3]
```diff
@@ -26,33 +26,23 @@ jobs:
...
-          newArtifactPath="${oldArtifactPath%'-signed.apk'}.apk"
-          echo "ARTIFACT_PATH=$newArtifactPath" >> $GITHUB_ENV
-          mv "$oldArtifactPath" "$newArtifactPath"
+          if [ ! -z "${{ secrets.KEYSTORE }}" ]; then
+            echo storePassword='${{ secrets.KEYSTORE_PASSWORD }}' >> keystore.properties
+            echo keyAlias='${{ secrets.KEY_ALIAS }}' >> keystore.properties
+            echo keyPassword='${{ secrets.KEY_PASSWORD }}' >> keystore.properties
+            echo storeFile='${{ github.workspace }}/key.jks' >> keystore.properties
+            echo ${{ secrets.KEYSTORE }} | base64 --decode > ${{ github.workspace }}/key.jks
+          fi
```
Comment (Goooler) [inline .github/workflows/build_debug_apk.yml:38]: Please verify these variables are right.

**Review by msfjarvis — APPROVED** [review]

**Comment by msfjarvis** [comment]
Thanks a lot for working on this!

**Comment by Goooler** [comment]
@msfjarvis If we can put source sets into their default locations?

**Comment by msfjarvis** [comment]
> @msfjarvis If we can put source sets into their default locations?

We can but it'll become a problem when we eventually attempt to merge the next release of Launcher3. Keeping Lawnchair changes minimal and isolated helps reduce merge conflicts during these updates.

File: build.gradle [hunk 4]
```diff
@@ -157,7 +151,10 @@ android {
         variant.outputs.all {
             outputFileName = "Lawnchair ${variant.versionName}.apk"
         }
-        resValue("string", "launcher_component", "${applicationId}/app.lawnchair.LawnchairLauncher")
```
Comment (Goooler) [inline build.gradle:160]: Fixed in #3121.

**Pull request merged**